packages/starknet-snap/src/ui/components/ExecuteTxnUI.tsx

```diff
@@ -0,0 +1,168 @@
+
+  // Sum up each call's transfer amount, converting to float
+  calls.forEach((call) => {
+    if (call.isTransfer) {
+      if (!tokenTotals[call.tokenSymbol]) {
+        tokenTotals[call.tokenSymbol] = { amount: 0, decimals: call.decimals };
+      }
+      tokenTotals[call.tokenSymbol].amount += parseFloat(call.amount);
+    }
+  });
+
+  // Convert maxFee (BigNumber) to float and add to feeToken's total if it's present
+  const feeTokenAmount = parseFloat(
+    formatUnits(maxFee, DEFAULT_DECIMAL_PLACES),
+  );
+  if (feeToken) {
+    if (tokenTotals[feeToken] === undefined) {
+      tokenTotals[feeToken] = {
+        amount: feeTokenAmount,
+        decimals: DEFAULT_DECIMAL_PLACES,
+      };
+    } else {
+      tokenTotals[feeToken].amount += feeTokenAmount;
+    }
+  }
+
+  return (
+    <Container>
+      <Box>
+        <Section>
+          <AddressUI label="Signer" address={signer} chainId={chainId} />
+        </Section>
+
+        {/* Loop through each call and render based on isTransfer */}
+
+        {calls.map((call) => (
+          <Section>
+            <AddressUI
+              label={call.label}
+              address={call.contractAddress}
+              chainId={chainId}
+              svgIcon={call.svgIcon}
+            />
+            {call.isTransfer ? (
+              <Section>
+                {call.senderAddress === signer ? null : (
+                  <AddressUI
+                    label="Sender Address"
+                    address={call.senderAddress}
+                    chainId={chainId}
+                  />
+                )}
+                <AddressUI
+                  label="Recipient Address"
+                  address={call.recipientAddress}
+                  chainId={chainId}
+                />
+                <RowUI
+                  label={`Amount`}
+                  value={`${call.amount as string} ${
+                    call.tokenSymbol as string
+                  }`}
+                />
+              </Section>
+            ) : (
+              <JsonDataUI label="Call Data" data={call.calldata} />
+            )}
+          </Section>
+        ))}
+        <FeeTokenSelector selectedToken={feeToken as FeeToken} />
+        <Section>
+          <Box direction="horizontal">
+            <Icon name="gas" size="md" />
+            <Row label="Estimated network fee">
+              <Value
+                value={`${formatUnits(
+                  maxFee,
+                  DEFAULT_DECIMAL_PLACES,
+                )} ${feeToken}`}
+                extra={`$0`}
+              />
+            </Row>
+          </Box>
+          <Box direction="horizontal">
+            <Icon name="money" size="md" />
+            <Box direction="vertical">
+              {Object.entries(tokenTotals).map(([tokenSymbol, { amount }]) => (
+                <Row key={tokenSymbol} label={`Total for ${tokenSymbol}`}>
+                  <Value value={`${amount} ${tokenSymbol}`} extra={`$0`} />
```

Comment:
double check if it has formatted unit

Reply:
do we need "extra={`$0`}" , we have not yet implement any price api yet 

remove it all until we implement price api